### PR TITLE
[GPR] Factorize Color ramp shader description

### LINKED ISSUE #3598: Use a link to raster pseudocolor section in the Mesh "Contour symbology" section
refs https://github.com/qgis/QGIS-Documentation/pull/3572/files#r268555990

The "Contour symbology" section in the mesh properties chapter seems highly inspired by the raster [singleband pseudocolor rendering](https://docs.qgis.org/testing/en/docs/user_manual/working_with_raster/raster_properties.html#singleband-pseudocolor) and if the two features are the same as I suspect, I suggest to avoid repetition using a reference to the raster section and only keep the mesh own features there (if any).
We could then improve the raster section with what is better described from the mesh side (and all this would be easy to maintain).

### PR DESCRIPTION
**Pull request by DelazJ**
This is the first of a series... let's call it `Gruyère Pull Requests (GPR)` because it has parts I'm unable to fill due to various reasons (no report available or if available not user friendly, not enough background on mesh thingies from myself,...). I really hope they will get enough feedback and completion for merging.

This PR introduces the color ramp shader widget available as such or through its settings in various raster/mesh symbology dialogs. Instead of describing in each dialog, let's put the description in one place and link to it. And try to explain the options instead of just listing the GUI. Fixes #3598
The PR is expected to fail as it relies on #6841

### CODE REVIEW

**Comment by DelazJ** [comment]
Not a single feedback on all these related pull requests...!?! This repo has no more reviewers?  😞 
@vcloarec @PeterPetrik (as features authors) May I have your deep review on at least this [set of pull requests](https://github.com/qgis/QGIS-Documentation/pulls?q=is%3Apr+is%3Aopen+%5BGPR%5D)? Thanks.

File: docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 1]
```diff
@@ -220,27 +220,6 @@ options of contours for the selected group, as shown in
 
    Styling Contours in a Mesh Layer
 
-Use the slide bar or combo box to set the opacity of the current group.
```
Comment (PeterPetrik) [inline docs/user_manual/working_with_mesh/mesh_properties.rst:223]: you can still colour contours in mesh layer, shall we have a link here for the same text you added for rasters?

File: docs/user_manual/working_with_raster/raster_properties.rst [hunk 2]
```diff
@@ -413,6 +367,79 @@ on the:
...
+   The name of the color ramp will be saved in the configuration.
+#. The :guilabel:`Label unit suffix` adds a label after the value in
+   the legend, and the :guilabel:`Label precision` controls the number of
+   decimals to display.
+#. Under the classification :guilabel:`Mode`, you can apply:
+
+   * :guilabel:`Equal interval`: you set a :guilabel:`number of classes`
+     and QGIS assigns their limits so that they all have the same magnitude
+   * :guilabel:`Continuous`: ???
+   * :guilabel:`Quantile`: you set a :guilabel:`number of classes` and
```
Comment (PeterPetrik) [inline docs/user_manual/working_with_raster/raster_properties.rst:411]: quantile is not available for meshes

**Comment by PeterPetrik** [comment]
Maybe @saberraz can be help here too: https://github.com/qgis/QGIS-Documentation/pulls?q=is%3Apr+is%3Aopen+%5BGPR%5D

File: docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 1 above)
Comment (DelazJ) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:223]: Hi @PeterPetrik 
Which text are you referring to?
In this PR I simply refactor the color ramp shader description. Do https://github.com/qgis/QGIS-Documentation/pull/6845/files#diff-977897f8087e85753e52d0d354623eb4bbdf6420f824dd64ff75039c3b5bb1d9R226 and subsequent correspond to what you are asking for?

File: docs/user_manual/working_with_raster/raster_properties.rst (hunk 2 above)
Comment (DelazJ) [reply docs/user_manual/working_with_raster/raster_properties.rst:411]: Will mention it. thanks for the reminder

**Comment by DelazJ** [comment]
> Maybe @saberraz can be help here too: https://github.com/qgis/QGIS-Documentation/pulls?q=is%3Apr+is%3Aopen+%5BGPR%5D

I'd love to. And please, do not hesitate to suggest rewording. It makes things much easier to proceed than \me trying to reformulate your explanation. TIA

File: docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 1 above)
Comment (PeterPetrik) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:223]: It looked like you removed the text from the mesh layer contour styling.

Comment (DelazJ) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:223]: Yes and no. I removed the text here but did not put replacement in this PR to make us focus on the color ramp shader description and because I was trying to split my work on mesh into smaller, self-contained and easy to review PRs. The updates of this section is in the link I provided above and now published in [testing docs](https://docs.qgis.org/testing/en/docs/user_manual/working_with_mesh/mesh_properties.html#contours-symbology) (vs [3.16](https://docs.qgis.org/3.16/en/docs/user_manual/working_with_mesh/mesh_properties.html#contours-symbology)). Do you miss something?

Comment (PeterPetrik) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:223]: looks good, thanks
